### PR TITLE
Unified api documentation generation

### PR DESCRIPTION
**Pull request by mjafshar**
Modifying grape_doc gem to create documentation for multiple APIs.

Our APIs will need to use this version of the grape_doc gem.

### CODE REVIEW

File: lib/grape_doc.rb [hunk 1]
```diff
@@ -11,11 +12,52 @@
 module GrapeDoc
   def self.generate_doc
     opts = Trollop::options do
-      opt :path, "Resource path", 
-          :type => :string, 
-          :default => File.expand_path(Dir.pwd + "/config/environment") 
+      opt :batch, "Generate documentation for multiple APIs",
+          :short => :B
```
Comment (alkalin3) [inline lib/grape_doc.rb:17]: Is there a reason you're using hash rockets instead of short: :B ?

File: lib/grape_doc.rb [hunk 2]
```diff
@@ -11,11 +12,52 @@
...
+          :short => :F,
+          :default => ["./config/environment.rb"]
+      opt :root_api, "Top level API class name",
+          :type => :string
+      opt :stdout, "Print documentation stdout instead of generating files",
+          :short => :s,
+          :default => false
+    end
+
+    FileUtils.mkdir_p(opts[:output_dir]) if !opts[:stdout]
```
Comment (alkalin3) [inline lib/grape_doc.rb]: This is a good use case for unless. It'll make the code a tad bit more readable.

If you don't like unless, I also prefer

``` ruby
 FileUtils.mkdir_p(opts[:output_dir]) if opts[:stdout].nil?
```

File: lib/grape_doc.rb (hunk 1 above)
Comment (mjafshar) [inline lib/grape_doc.rb:17]: I was just following the pattern the original developer started.

Comment (alkalin3) [inline lib/grape_doc.rb:17]: That makes sense to me.

**Comment by alkalin3** [comment]
This looks pretty solid dude.

**Pull request merged**